modules/gitops-argo-rollouts-architecture-overview.adoc

```diff
@@ -0,0 +1,70 @@
+|`Services / Ingresses` | Rollouts natively support routing traffic by Services and Ingresses, using the Service and Ingress controllers.
+|`OpenShift Route resources` | Rollouts perform traffic splitting across different application versions using OpenShift Route resources.
+|`OpenShift Service Mesh's VirtualService resources` | Rollouts also perform traffic splitting across different versions of an Application, using OpenShift Service Mesh with Gateways VirtualServices resources.
+|===
+
+Argo Rollouts include several mechanisms for gathering analysis metrics for verifying that a new version of an application is deployed.
+
+[options="header"]
+|===
+|Name | Description
+|`Prometheus metrics` | `AnalysisTemplate` is configured to connect to Prometheus instances, to evaluate the success or failure of one or more metrics.
+|`Kubernetes Job resource metrics` | Rollouts support the use of simple Kubernetes jobs to run analysis on resource metrics. You can verify a successful deployment of an application based on the successful execution of Kubernetes jobs.
+|===
+
+[id="gitops-components_{context}"]
+== Components
+
+Argo Rollouts consists of several components that enable users to practice Progressive Delivery on OpenShift.
+
+[options="header"]
+|===
+|Name | Description
+|`Rollouts Controller` | The Rollouts Controller component of the Argo Rollouts manages the Rollouts custom resource which resembles a native Kubernetes Deployment resource. The Rollouts Controller does not modify the standard deployment resources and only responds to changes in the Rollouts resources.
+|`AnalysisRun Controller` | The `AnalysisRun` controller performs analysis for an `AnalysisRun` resource and manages the `AnalysisRun` custom resource. It provides the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that decide if a deployment update is successful for your application.
+
+Argo Rollouts supports two custom Kubernetes resources to perform Analysis - `AnalysisTemplate` and `AnalysisRun`. You can define an `AnalysisTemplate` on a specific Rollout or globally on a cluster to be shared by multiple rollouts as a Cluster`AnalysisTemplate`. The `AnalysisRun` resource is scoped on a specific rollout.
+|`Experiment Controller` | The `Experiment` Controller creates analysis for limited-duration, short-lived replica sets, and manages the `Experiment` custom resource. The `Experiment` custom resource can be used independently or executed as part of Rollouts. You can link a list of `AnalysisTemplate` to be used on replica sets by creating an `Experiment` custom resource.
+
+In addition, this controller is used for creating replica sets. You can use it for optionally creating the service, running the analysis, reporting the results, and scaling down the replica sets in the end. The `Experiment` controller can also be integrated with Rollouts by specifying the experiment step in the canary deployment strategy.
```

Comment:
```suggestion
In addition, this controller is used to create these replica sets. Optionally, you can use it to create a service, run the analysis, report the results, and scale down the replica sets. The controller can also be integrated with `Rollout` resource by specifying the `experiment ` step in the canary deployment `strategy` field .
```
1) The sentence: "You can use it for optionally creating the service, running the analysis,..."
Isn't the whole point to run an analysis here?  Why is it marked as optional? This is all a little confusing to me.

2) I suggested the following: "The controller can also be integrated with the `Rollout` resource by specifying the `experiment ` step in the canary deployment `strategy` field ."  I think this is clearer to understand, but I just want to check if my perceived meaning is correct :+1:

Reply:
Fixed.

Reply:
@jgwest - FYI

Reply:
For the second paragraph, I recommend we just remove the first/second sentence and move up the remaining sentence to the paragraph above.
e.g.:
~In addition, this controller is used to create these replica sets. Optionally, you can use it to create a service, run the analysis, report the results, and scale down the replica sets.~ The controller can also be integrated with `Rollout` resource by specifying the `experiment ` step in the canary deployment `strategy` field .

I don't think the first sentence of the second paragraph includes any content that is needed.